hazelcast/src/test/java/com/hazelcast/map/IndexStatsChangingNumberOfMembersTest.java

```diff
@@ -409,7 +418,36 @@ protected double calculateOverallSelectivity(long initialHits, double initialTot
     }
 
     protected void addIndex(IMap<?, ?> map) {
-        map.addIndex(new IndexConfig(IndexType.HASH, "this").setName("this"));
+        map.addIndex(new IndexConfig(IndexType.HASH, "this").setName(INDEX_NAME));
     }
 
+    protected void awaitStable(String mapName, HazelcastInstance... instances) {
+        // Await for migrations to complete.
+        waitAllForSafeState(instances);
+
+        // Make sure that all indexes contain expected partitions.
+        Map<UUID, PartitionIdSet> memberToPartitions = new HashMap<>();
+
+        Set<Partition> partitions = instances[0].getPartitionService().getPartitions();
+
+        for (Partition partition : partitions) {
```

Comment:
Should we not reevaluate this UUID to partition mapping in the `assertTrueEventually` block? I'm asking because I can imagine that this was invoked with the 1st member being the master member while the 0th member might have had a stale partition table. We then wait until all members have the same partition table as the 0th, which might not happen.

Reply:
This additional check is invoked after `waitAllForSafeState`, so the partition distribution is expected to be stable at this point.

Reply:
Good point, thanks!